016-05-02  Ziheng Yang  <zihengy@lnxsrv02.seas.ucla.edu>

        * ls.c: fixed issue with integer overflow when ordering files by date

--- coreutils-with-bug/src/ls.c	2016-04-28 13:51:27.622265000 -0700
+++ old-coreutils/coreutils-with-bug/src/ls.c	2005-10-15 02:56:21.000000000 -0700
@@ -2877,15 +2877,8 @@
 cmp_mtime (struct fileinfo const *a, struct fileinfo const *b,
 	   int (*cmp) (char const *, char const *))
 {
-  int diff = 0;
-  if ((get_stat_mtime (&b->stat)).tv_sec < 0 && (get_stat_mtime(&a->stat)).tv_sec > 0)
-    diff = -1;
-  else if ((get_stat_mtime (&b->stat)).tv_sec > 0 && (get_stat_mtime(&a->stat)).tv_sec < 0)
-    diff = 1;
-  else{
-    diff = timespec_cmp (get_stat_mtime (&b->stat),
-			     get_stat_mtime (&a->stat));
-  }
+  int diff = timespec_cmp (get_stat_mtime (&b->stat),
+			   get_stat_mtime (&a->stat));
   return diff ? diff : cmp (a->name, b->name);
 }
 static int compare_mtime (V a, V b) { return cmp_mtime (a, b, xstrcoll); }
